### PR TITLE
[RFR] Use admin-on-rest

### PR DESCRIPTION
**Pull request by djhi**
- [x] Added basic admin configuration for products and orders
- [x] Added a custom rest client with jwt token
- [x] Added products in order creation and edition form

screen
screen

### CODE REVIEW

**Comment by fzaninotto** [comment]
Very encouraging. It's so straightforward now!

**Comment by djhi** [comment]
Remaining problems:

I have to declare a route and related components for order-products. See https://github.com/marmelab/admin-on-rest/issues/18

**Comment by djhi** [comment]
I'll see what I can do with a custom Layout as discussed

File: makefile [hunk 1]
```diff
@@ -82,6 +82,11 @@ run-dev: ## Run all applications in development environment (using webpack-dev-s
 	@echo "  Admin app:    http://localhost:8080/admin"
 	@echo "Type 'make stop-dev' to stop the apps"
 
+restart-dev: ## Stop all applications in development environment
```
Comment (fzaninotto) [inline makefile]: s/Stop/Restart/

Comment (djhi) [inline makefile]: This sedy bot is awesome :) :+1: @Kmaschta

File: src/admin/js/Layout.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+const Layout = ({ isLoading, children, route }) => {
+    const Title = <Link to="/" style={{ color: '#fff', textDecoration: 'none' }}>{`${APP_NAME} - Administration`}</Link>;
+    const RightElement = isLoading ? <CircularProgress color="#fff" size={0.5} /> : <span />;
+
+    return (
+        <MuiThemeProvider>
+            <div style={{ display: 'flex', flexDirection: 'column', minHeight: '100vh' }}>
+                <AppBar title={Title} iconElementRight={RightElement} />
+                <div className="body" style={{ display: 'flex', flex: '1', backgroundColor: '#edecec' }}>
+                    <Menu />
```
Comment (fzaninotto) [inline src/admin/js/Layout.js]: You can put the `<Menu>` in the next line to make the content reads first in the source. The `flex` property insures it's displayed first.

File: src/admin/js/user/sagas.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+export const signOut = removeLocalUser => function* signOutSaga() {
+    yield call(removeLocalUser);
+    yield put(signOutActions.success());
+    yield put(routerActions.push('/'));
+};
+
+const sagas = function* sagas() {
+    yield [
+        takeLatest(userActionTypes.signIn.REQUEST, signIn(fetchSignInApi, storeLocalUserApi)),
+        takeLatest(userActionTypes.signUp.REQUEST, signUp(fetchSignUpApi, storeLocalUserApi)),
```
Comment (fzaninotto) [inline src/admin/js/user/sagas.js]: Isn't it the occasion to benefit from saga, and `yield` `signOut` only after yielding `singIn`?

http://yelouafi.github.io/redux-saga/docs/advanced/NonBlockingCalls.html

File: webpack/loaders.js [hunk 4]
```diff
@@ -7,13 +7,17 @@ export default function (appName) {
 
     const loaders = [{
         test: /\.jsx?$/,
-        exclude: /node_modules/,
+        exclude: /node_modules\/(?!(admin-on-rest))/,
```
Comment (fzaninotto) [inline webpack/loaders.js]: should ne be needed

File: src/admin/js/Admin.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import Layout from './Layout';
+import { OrderList, OrderEdit, OrderCreate } from './orders';
+import { OrderProductList, OrderProductEdit } from './order-products';
+import { ProductList, ProductEdit, ProductCreate } from './products';
+
+// create a Redux app
+const reducer = combineReducers({
+    admin: adminReducer([{ name: 'products' }, { name: 'orders' }, { name: 'order-products' }]),
+    routing: routerReducer,
+});
```
Comment (fzaninotto) [inline src/admin/js/Admin.js]: shouldn't your `user` reducer be included here, too?

File: src/admin/js/user/sagas.js (hunk 3 above)
Comment (djhi) [inline src/admin/js/user/sagas.js]: well it's not used yet anyway, I forgot about it. back to wip

**Comment by djhi** [comment]
Not happy yet with the layout & loading icon but it works

File: makefile [hunk 6]
```diff
@@ -14,7 +14,7 @@ PM2_HOME ?= .pm2
 help:
 	@grep -P '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
 
-# Initialization ===============================================================
+#B Initialization ===============================================================
```
Comment (Kmaschta) [inline makefile]: B?

File: package.json [hunk 7]
```diff
@@ -31,64 +31,66 @@
     "koa-route": "2.4.2",
     "lodash._baseflatten": "4.2.1",
     "lodash.pick": "4.4.0",
+    "material-ui": "0.15.4",
```
Comment (Kmaschta) [inline package.json]: Why `material-ui` is in dependencies if `admin-on-rest` is on dev dependencies?

Comment (djhi) [inline package.json]: Because we also use it directly

Comment (djhi) [inline package.json]: oh right, admin-on-rest should be in deps now

**Pull request merged**